src/controller/CHIPDeviceController.cpp

```diff
@@ -1926,6 +1915,8 @@ void DeviceCommissioner::OnDeviceConnectionRetryFn(void * context, const ScopedN
                  ChipLogValueScopedNodeId(peerId), error.Format(), retryTimeout.count());
 
     auto self = static_cast<DeviceCommissioner *>(context);
+    VerifyOrDie(self->mCommissioningStage == CommissioningStage::kFindOperational);
+    VerifyOrDie(self->mDeviceBeingCommissioned->GetDeviceId() == peerId.GetNodeId());
```

Comment:
We are certain this cannot ever be false at runtime, in a recoverable way, right?

Reply:
Yes, we're invalidating the CASE callbacks when we're done with them and on StopPairing now, so there should be no way to get stray callbacks anymore.